### ISSUE TITLE
Collection Change Listener stops listening until next event emits and then proceeds to stop listening

### ISSUE THREAD

**Issue by NqCompiler** [realm/realm-js#4708]
### How frequently does the bug occur?

Sometimes

### Description

Our application starts initializing realm when mounting the main screen.
It starts listening to ~5 different collections.
Initially everything seems to be working fine.
However, after for example, refreshing/reloading the app (debug mode), only the initial callbacks are called for the collection listeners but they don't react to any changes.
If I make emit any event (a realm.write transaction or net-info) the change listeners are reacting again but proceed to stop listening.

If I call the initialization of realm two times - directly after mount & then couple of seconds later again (closing the previously initialized realm instance) - the problem occures less often.
I tried different devices and with mobiles on better hardware the problem occurs more often.
This happens also in PRODUCTION.
Maybe it has something to do with react-native itself, but maybe you can lead me to the root of the problem as I experience this bug only when using realm.

### Can you reproduce the bug?

Yes, sometimes

### Version

10.20.0-beta.3, 10.20.0-beta.5

### What SDK flavour are you using?

Atlas App Services (auth, functions, etc.)

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

ANDROID

### Build environment

"react-native": "0.67.4"
"realm": "^10.20.0-beta.5"

**Comment by NqCompiler** [comment]
I have come across this issue as well but there are some differences:
- touching the UI does not trigger an update
- setting state within the callback function does not work as temporary solution because the callback is not triggered at all

but it is also possible it might be a variant...
I will leave it open and when #2655 is fixed I'll check if it works for me

**Comment by NqCompiler** [comment]
I can't use the chrome debugger (Error: This version of Realm JS doesn't support the legacy Chrome Debugger. Please use Flipper instead. see https://github.com/realm/realm-js/issues/4364 )

So I assume I am already running in on "non-debug" mode (?), if not can I toggle it via the RN "In-App Developer Menu"?
please note: this happens in production

**Comment by NqCompiler** [comment]
I managed to rewrite our code and the error is now  #2655 

For your information:
After implementing @realm/react, the error still appeared.
Our application uses two realm instances but not at the same time.
First realm instance starts initially and after unmounting the screen, it is closed. Then the second instance is opened.
We rewrote this code by using one realm instance and voila we arrived at #2655 which is much easier to handle or accept as a bug.

Thanks for the help
